src/test/compile-fail/issue-68060.rs

```diff
@@ -0,0 +1,14 @@
+#![feature(track_caller)]
```

Comment:
Any specific reason to put this test into `compile-fail` instead of `ui`?
`compile-fail` is generally deprecated and only contains tests with not fully reproducible or heavily platform-specific output.

Reply:
It requires _running_ test, not just checking to trigger ICE with `#[target_feature]`. I'm not familiar with running test with errors, any idea?

Reply:
I don't understand, this particular test requires running or `ui` tests require running?

Reply:
Ah sorry, this test requires running code.

Reply:
Why?

Reply:
To reproduce the ICE requires running code. build, check, or something doesn't show it, you can see [playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=54de2f854ac5cef143677ea3958047a1). I usually use `run-pass` for such this case but we cannot use here since the test has some errors. Also `run-fail` cannot be also used since the compilation will be failed. I confirmed the ICE isn't shown even without the fix when I place this test in `ui`. That's why I placed it in `compile-fail`. But I'm not familiar with ui-test fully so I may be missing something. Would you have any practice?

Reply:
I'd recommend to trust playground less :)
Looks like it's actually doing `cargo check` on "Build".
Regular `rustc` produces an error when building this code.

UI tests also test in `check` mode by default, you have to add `// build-fail` to them to catch later codegen-time or link-time failures.

Reply:
Oh, thanks! `build-fail` works well. Moved the test.